plugins/chat/spec/models/chat/channel_spec.rb

```diff
@@ -11,6 +11,14 @@
   it { is_expected.to validate_length_of(:chatable_type).is_at_most(100) }
   it { is_expected.to validate_length_of(:type).is_at_most(100) }
 
+  it "supports custom fields" do
```

Comment:
Instead of repeating that same example over and over, we already have a shared example for custom fields. So we can replace this by `it_behaves_like "it has custom fields"` :slightly_smiling_face: 

For channel specs, instead of putting it in the base model spec, it’s better to have it in each child class (`Chat::CategoryChannel` and `Chat::DirectMessageChannel`), that way we ensure there’s no problem with the real concrete classes.

Then it’s just a matter of adding the new example to the existing shared example, I also had to make some adjustments for the shared example to work as expected:
```ruby
# in spec/support/shared_examples_for_custom_fields.rb
# instead of  let(:record) { described_class.new } on line 4
  let(:fabricator_name) do
    described_class
      .name
      .underscore
      .tr("/", "_")
      .then do |name|
        next name if Fabrication.manager[name]
        name.sub("chat_", "")
      end
  end
  let(:record) { Fabricate.build(fabricator_name) }

# then at the end of the shared example
  describe "Saving custom fields" do
    let(:record) { Fabricate(fabricator_name) }

    it "supports saving custom fields" do
      record.custom_fields["test"] = "test"
      expect { record.save_custom_fields }.to change {
        described_class.find(record.id).custom_fields["test"]
      }.to "test"
    end
  end
```
I have a commit with all the necessary changes ready to be pushed to your branch if you prefer, though :grin:

Reply:
I really really really really really dislike shared spec, I find it very hard to debug, much prefer seeing exactly what is happening in the spec vs needing to juggle through multiple files to figure out why something is working or not.